noccur.el

```diff
@@ -54,13 +54,26 @@ ls-files' and 'grep'."
   (interactive (occur-read-primary-args))
   (let* ((default-directory (read-directory-name "Search in directory: "))
          (files (mapcar #'find-file-noselect
-                        (noccur--find-files regexp))))
+                        (noccur--find-files-project regexp))))
+    (multi-occur files regexp nlines)))
+
+;;;###autoload
+(defun noccur-directory (regexp &optional nlines)
```

Comment:
I've googled what are the elisp's most common way of naming this kind of things. What I've found is that when this is recursive, that keyword "recursive" is part of the function name.
So I would propose as proposition 1:
- `noccur-directory` --> `noccur-directory`
- `noccur-project` --> `noccur-directory-recursive`

As per the anecdote, I've found `noccur`, because I was looking for a `multi-occur` for files in a directory. I found strange, and did not understood at first why `noccur-project` was named like that, with "project" in its name. I understood later that this was because of projectile. 

I could also propose as proposition 2:
- `noccur-project` (recursive as much as the scope of the git project, falling back to `find` if no git/bzr/hg)
- `noccur-directory-recursive` (not using `git ls-files`)
- `noccur-directory` (not using `git ls-files`, not recursive)

I've just realized that my home directory `~` is under revision control with git. Here is content of my `.gitignore` file:

    /*
    !/.emacs

That means that `git ls-files` is only list one file, my `.emacs`. So `noccur-project` or `noccur-directory`, in their current form will not work for me.

This is another argument to proposition 2. That is creating `noccur-directory-*`, not using `git ls-files`, only using `find`. 

We will loose the `git ls-files` of depth 1, but this is an uncommon use-case. `find -maxdepth 1` is a more frequent use-case.

If you agree, I will modify this pull request to be aligned with proposition 2.